packages/costa/src/runtime.ts

```diff
@@ -0,0 +1,265 @@
+    this.assignPackage(pkg, source);
+    return pkg;
+  }
+  /**
+   * Install the given plugin by name.
+   * @param name the plugin package name.
+   * @param force install from new anyway.
+   * @param pyIndex the index mirror to install python packages.
+   */
+  async install(pkg: PluginPackage, force = false, pyIndex?: string): Promise<boolean> {
+    // check if the pkg is installed
+    if ((await this.isInstalled(pkg.name)) && !force) {
+      debug(`skip install "${pkg.name}" because it already exists`);
+      return true;
+    }
+
+    const pluginStdName = `${pkg.name}@${pkg.version}`;
+    let pluginAbsName;
+    if (pkg.pipcook.source.from === 'npm') {
+      pluginAbsName = pluginStdName;
+      debug(`install the plugin from npm registry: ${pluginAbsName}`);
+    } else {
+      pluginAbsName = pkg.pipcook.source.uri;
+      debug(`install the plugin from local: ${pluginAbsName}`);
+    }
+    await spawnAsync('npm', [
+      'install', `${pluginAbsName}`, '--no-save'
+    ], {
+      cwd: this.options.installDir
+    });
+
+    if (pkg.conda?.dependencies) {
+      debug(`prepare the Python environment for ${pluginStdName}`);
+      const envDir = path.join(this.options.installDir, 'conda_envs', pluginStdName);
+      await remove(envDir);
+      await ensureDir(envDir);
+
+      debug(`install the Python environment for ${pluginStdName}`);
+      const requirements = createRequirements(pluginStdName, pkg.conda);
+      // just sync the copy to requirements.txt
+      await writeFile(
+        `${envDir}/requirements.txt`,
+        requirements.join('\n')
+      );
+
+      let python;
+      try {
+        const condaInstallDir = await readFile(CONDA_CONFIG, 'utf8');
+        python = `${condaInstallDir}/bin/python3`;
+        await access(python);
+      } catch (err) {
+        debug(`occuring an error when fetching conda: ${err && err.stack}`);
+        throw new Error('Invalid boa/conda installation, please try to install.');
+      }
+
+      debug('conda environment is setup correctly, start downloading.');
+      await spawnAsync(python, [ '-m', 'venv', envDir ]);
+      // TODO(yorkie): check for access(pip3)
+
+      for (let name of requirements) {
+        debug(`installing python package ${name}`);
+        let args = [ 'install', name ];
+        if (pyIndex) {
+          args = args.concat([ '-i', pyIndex ]);
+        }
+        await spawnAsync(`${envDir}/bin/pip3`, args);
+      }
+    } else {
+      debug(`just skip the Python environment installation.`);
+    }
+    
+    await this.linkBoa();
+    return true;
+  }
+  /**
+   * Uninstall the given plugin by name.
+   * @param name the plugin package name.
+   */
+  async uninstall(name: string): Promise<boolean> {
+    if (!await this.isInstalled(name)) {
+      debug(`skip uninstall "${name}" because it not exists.`);
+      return false;
+    }
+    await remove(path.join(this.options.installDir, 'node_modules', name));
+    return true;
```

Comment:
Using `npm uninstall <package-name>` will be better, remove won't purge its dependencies at the same time.

Reply:
Actually I used `npm uninstall`, but it will cleanup `node_modules` under the plugin directory.